plugins/@grouparoo/airtable/src/lib/destination/exportRecords.ts

```diff
@@ -0,0 +1,322 @@
+    tableId,
+    primaryKey,
+    foreignKeys
+  );
+  records.forEach((record) => {
+    const key = record.fields[primaryKey.toString()];
+    if (!key) {
+      return;
+    }
+    const found = getByForeignKey(key.toString());
+    if (found) {
+      found.destinationId = record.id;
+      found.result = record.fields;
+    }
+  });
+};
+
+function buildUpdatePayload(
+  exportedProfile: BatchExport
+): RecordData<Partial<FieldSet>> {
+  if (!exportedProfile.destinationId) {
+    throw ReferenceError(
+      `Could not find destination ID in Batch Export with key ${exportedProfile.recordId}`
+    );
+  }
+  return {
+    id: exportedProfile.destinationId,
+    fields: buildPayloadFields(exportedProfile),
+  };
+}
+
+function buildCreatePayload(
+  exportedProfile: BatchExport
+): CreateRecord<FieldSet> {
+  return {
+    fields: buildPayloadFields(exportedProfile),
+  };
+}
+
+function buildPayloadFields(exportedProfile: BatchExport): FieldSet {
+  const { oldRecordProperties, newRecordProperties } = exportedProfile;
+  const deletePropertiesPayload = {};
+  const newPropertyKeys = Object.keys(newRecordProperties);
+  Object.keys(oldRecordProperties)
+    .filter((k) => !newPropertyKeys.includes(k))
+    .forEach((k) => (deletePropertiesPayload[k] = null));
+
+  let payload = Object.assign(newRecordProperties, deletePropertiesPayload);
+  const formattedDataFields = {};
+  for (const key of Object.keys(payload)) {
+    formattedDataFields[key] = formatVar(payload[key]);
+  }
+  return formattedDataFields;
+}
+
+function formatVar(value) {
+  if (value === null || value === undefined) {
+    return null; // empty string clears the value
+  }
+  if (value instanceof Date) {
+    return value.toISOString();
+  }
+  return value;
+}
+
+const addToGroups: BatchMethodAddToGroups = async () => {
+  // This destination doesn't do groups
+};
+
+const removeFromGroups: BatchMethodRemoveFromGroups = async () => {
+  // This destination doesn't do groups
+};
+// mess with the keys (lowercase emails, for example)
+const normalizeForeignKeyValue: BatchMethodNormalizeForeignKeyValue = ({
+  keyValue,
+}) => {
+  if (!keyValue) {
+    return null;
+  }
+  // Formula search is case sensitive
+  return keyValue.toString().trim();
+};
+
+// mess with the names of groups (tags with no spaces, for example)
+const normalizeGroupName: BatchMethodNormalizeGroupName = ({ groupName }) => {
+  return groupName.toString().trim();
+};
+
+interface ExportBatchOptions {
+  appOptions: SimpleAppOptions;
+  connection: IClient;
+  destinationOptions: SimpleDestinationOptions;
+  syncOperations?: DestinationSyncOperations;
+  exports: BatchExport[];
+}
+
+export async function exportBatch(exportBatchOptions: ExportBatchOptions) {
+  const { syncOperations, appOptions, destinationOptions, exports } =
+    exportBatchOptions;
+  const { primaryKey } = destinationOptions;
+  const batchSize = BATCH_SIZE;
+  const findSize = BATCH_SIZE;
```

Comment:
If these are constants from higher up, it's probably not nescisary to reassign them